documentation/src/main/asciidoc/topics/search.adoc

```diff
@@ -0,0 +1,1096 @@
+=== Rebuilding Indexes
+
+Rebuilding an index reconstructs it from data stored in the cache. You need to rebuild indexes if you change things like
+definitions of indexed types or Analyzers. Likewise you might need to rebuild indexes if they are deleted for some reason.
+Beware it might take some time as it needs to reprocess all data in the grid!
+
+[source,java]
+----
+Indexer indexer = Search.getIndexer(cache);
+CompletionStage<Void> future = index.run();
+----
+
+//TODO
+//[[query_indexless]]
+//==== Indexless
+
+//TODO
+//[[query_hybrid]]
+//==== Hybrid
+
+== Searching
+
+Create relational and full-text queries in both Library and Remote Client-Server mode with the Ickle query language.
+
+To use the API, first obtain a QueryFactory to the cache and then call the `.create()` method, passing in the string
+to use in the query. Each _QueryFactory_ instance is bound to the same _Cache_ instance as the _Search_, but it is
+otherwise a stateless and thread-safe object that can be used for creating multiple queries in parallel.
+
+For instance:
+
+[source,java,tile="Using Ickle"]
+----
+// Remote Query, using protobuf
+QueryFactory qf = org.infinispan.client.hotrod.Search.getQueryFactory(remoteCache);
+Query q = qf.create("from sample_bank_account.Transaction where amount > 20");
+
+// Embedded Query using Java Objects
+QueryFactory qf = org.infinispan.query.Search.getQueryFactory(cache);
+Query q = qf.create("from com.acme.Book where price > 20");
+
+// Execute the query
+QueryResult<Book> queryResult = q.execute();
+----
+
+
+NOTE: A query will always target a single entity type and is evaluated over the contents of a single cache. Running a
+query over multiple caches or creating queries that target several entity types (joins) is not supported.
+
+Executing the query and fetching the results is as simple as invoking the `run()` method of the _Query_ object. Once
+executed, calling `run()` on the same instance will re-execute the query.
+
+=== Pagination
+
+You can limit the number of returned results by using  the `Query.maxResults(int maxResults)`. This can be used in
+conjunction with `Query.startOffset(long startOffset)` to achieve pagination of the result set.
+
+[source,java]
+----
+// sorted by year and match all books that have "clustering" in their title
+// and return the third page of 10 results
+Query<Book> query = queryFactory.create("FROM com.acme.Book WHERE title like '%clustering%' ORDER BY year").startOffset(20).maxResults(10)
+----
+
+=== Number of Hits
+
+The `QueryResult` object has the `.hitCount()` method to return the total number of results of the query, regardless
+of any pagination parameter. The hit count is only available for indexed queries for performance reasons.
+
+
+=== Iteration
+
+The `Query` object has the `.iterator()` method to obtain the results lazily. It returns an instance of `CloseableIterator` that
+must be closed after usage.
+
+NOTE: Iteration is not currently implemented for Remote Queries
```

Comment:
Consider saying what is supported, rather than what is not supported. Possibly: "Iteration is possible only when using {brandname} as an embedded library."

Reply:
This is already outdated, this feature is now "emulated"